### PR TITLE
Add constraint to model

### PR DESCRIPTION
**Pull request by lipwei1808**
Added constraints to validation checks in Model.
1. Maximum of 500 characters for `Name`
2. Maximum of 500 characters for `Location`
3. Maximum of `Integer.MAX_VALUE` for `Amount`

One note on `parseIndex` for values greater than `Integer.MAX_VALUE`, the parser correctly identifies and catches values greater than the limit however the error is caught by the Parser using it (`EditTransactionCommandParser`) and the custom command message constraint is returned to the user. Thus, the error of invalid integer is not fed back to the user. This might not be a bug

### CODE REVIEW

File: src/main/java/unicash/model/transaction/Name.java [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters and spaces up to 500 characters, "
+                    + "and it should not be blank";
 
     /*
      * The first character of the transaction must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "^[\\p{Alnum}][\\p{Alnum} ]{0,499}$";
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/transaction/Name.java:20]: I think the validation regex for names can be made as lenient as location information! I don't see a reason why we cannot support non-alnum characters in names

File: src/main/java/unicash/model/transaction/Location.java [hunk 2]
```diff
@@ -9,13 +9,13 @@
 public class Location {
     public static final String MESSAGE_CONSTRAINTS =
             "Locations should only contain alphanumeric characters, spaces, (, ), _, -, #, &, ., and ',', "
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/transaction/Location.java:11]:
```suggestion
            "Locations should only contain alphanumeric characters, spaces, (, ), _, -, #, @, &, ., and ',', "
```

File: src/main/java/unicash/model/transaction/Location.java [hunk 3]
```diff
@@ -9,13 +9,13 @@
...
-                    + "and it should not be blank";
+                    + "up to 500 characters and it should not be blank";
 
     /*
      * The first character of the location must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-]*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-]{0,499}";
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/transaction/Location.java:18]:
```suggestion
    public static final String VALIDATION_REGEX = "[\\p{Alnum}_&#.,()-][\\p{Alnum} _&#.,()-@]{0,499}";
```

Reason is, users might wanna input sub-locations i.e. `McDonalds @ Clementi`

File: src/main/java/unicash/model/transaction/Amount.java [hunk 4]
```diff
@@ -26,7 +26,7 @@ public Amount(double amount) {
      * Returns true if a given amount is a non-negative value.
      */
     public static boolean isValidAmount(double amount) {
-        return amount >= 0.00;
+        return amount >= 0.00 && amount <= Integer.MAX_VALUE;
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/transaction/Amount.java:29]: I think there might be a merge conflict here, I've added the validation methods you requested together with #96

**Review by sp4ce-cowboy — CHANGES_REQUESTED** [review]
LGTM, just some minor user input considerations

File: src/main/java/unicash/model/transaction/Location.java (hunk 3 above)
Comment (woojiahao) [reply src/main/java/unicash/model/transaction/Location.java:18]: Believe the `@` symbol must come before the `-` symbol

**Review by woojiahao — COMMENTED** [review]
Think there needs to be a change to the recommended addition of `@` but other than that, LGTM, are we going to display the error when the user input is greater than `Integer.MAX_VALUE`?

**Comment by lipwei1808** [comment]
> Think there needs to be a change to the recommended addition of `@` but other than that, LGTM, are we going to display the error when the user input is greater than `Integer.MAX_VALUE`?

I think we can add an extension to the message constraint for `Amount` to include up to Integer.MAX_VALUE? Or would that not be too intuitive for a user that is not a developer.

**Review by woojiahao — APPROVED** [review]
LGTM

**Review by sp4ce-cowboy — APPROVED** [review]
Looks good!

**Pull request merged**